### DISCUSSION TITLE
You can show only actual menu items for users.

### DISCUSSION THREAD

**Discussion by snenko** [YetiForceCompany/YetiForceCRM#14142] [Feature improvement]
![image](https://user-images.githubusercontent.com/11363876/106370587-13808000-6364-11eb-8829-977fcc499c41.png)

if you add **isShowTopLevelMenuItem** function into **Vtiger_Menu_Model** class:

`
public static function isShowTopLevelMenuItem($menu)
{
  if(isset($menu['childs']) && count($menu['childs'])>0) {
	foreach ($menu['childs'] as $id=>$subMenu) {

		$privilegesModel = Users_Privileges_Model::getCurrentUserPrivilegesModel();
		if(\App\Module::isModuleActive($subMenu['mod'])==false) {
			continue;
		}

		if ($privilegesModel->isAdminUser() ||
			$privilegesModel->hasGlobalReadPermission() ||
			$privilegesModel->hasModulePermission($subMenu['tabid']) ) {
			return true;
		}

	}
	return false;
  }
  return true;
}
`

And update TPL file:
**/layouts/basic/modules/Vtiger/menu/Label.tpl**
![image](https://user-images.githubusercontent.com/11363876/106370624-5e01fc80-6364-11eb-82cf-8be2202067d7.png)
`	{if $MENU_MODULE=='Settings::Vtiger'}
		{assign var=SHOWITEMMENU value='true'}
	{else}
		{assign var=SHOWITEMMENU value=Vtiger_Menu_Model::isShowTopLevelMenuItem($MENU, $MENU_MODULE)}
	{/if}

	{if $SHOWITEMMENU=='true'}

	<li class="tpl-menu-Label c-menu__item js-menu__item nav-item menuLabel {if !$HASCHILDS}hasParentMenu{/if}" data-id="{$MENU['id']}" data-js="mouseenter mouseleave">
		<a class="nav-link {if $ACTIVE=='true'}active{else}collapsed{/if}{if $ICON} hasIcon{/if}{if $HASCHILDS == 'true'} js-submenu-toggler is-submenu-toggler{/if}" href="#"
				{if $HASCHILDS == 'true'} data-toggle="collapse" data-target="#submenu-{$MENU['id']}" role="button" aria-haspopup="true" aria-expanded="{$ACTIVE}" aria-controls="submenu-{$MENU['id']}"{else} role="heading"{/if} >
			{$ICON}
			<span class="c-menu__item__text js-menu__item__text" title="{Vtiger_Menu_Model::vtranslateMenu($MENU['name'],$MENU_MODULE)}" data-js="class: u-white-space-n">{Vtiger_Menu_Model::vtranslateMenu($MENU['name'],$MENU_MODULE)}</span>
			{if $HASCHILDS == 'true'}
				<span class="toggler" aria-hidden="true"><span class="fas fa-plus-circle"></span><span class="fas fa-minus-circle"></span></span>
			{/if}
		</a>
		{include file=\App\Layout::getTemplatePath('menu/SubMenu.tpl', $MODULE)}
	</li>

	{/if}`

**Comment by snenko** [comment]
I try to implement this via custom override code in my project.
But I think "custom" override functionality is not working for layouts and models.
So, each Yetiforce's updates remove my changes from the core folder

**Reply by vovpff** [reply]
Hm. For 5.3 if module inactive for user profile then this module invisible in menu
